src/ai/backend/storage/weka/exceptions.py

```diff
@@ -0,0 +1,39 @@
+class WekaError(Exception):
+    pass
+
+
+class WekaInitError(WekaError):
+    pass
+
+
+class WekaAPIError(WekaError):
+    name = 'WekaAPIError'
+    message = ''
+
+    def __init__(self, message: str = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.message = message or 'Unknown Error'
+
+    def __str__(self) -> str:
+        return f'{self.name}: {self.message}'
```

Comment:
This could be replaced with `type(self).__name__` instead of duplicating exception names as string attributes.
```python
>>> class MyAwesomeError(Exception):
...   def __str__(self):
...     return type(self).__name__
...
>>> str(MyAwesomeError())
'MyAwesomeError'
```

Reply:
Also, if you utilize `self.args`, you could just use `repr()` to get a canonical string representation of the exception class with all its arguments.

```python
>>> class MyXXError(Exception):
...   pass
...
>>> x = MyXXError('abc', 123)
>>> str(x)
"('abc', 123)"
>>> repr(x)
"MyXXError('abc', 123)"
```